src/Microsoft.AspNetCore.Mvc.Core/RequestFormLimitsAttribute.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// Sets the specified limits to the <see cref="HttpRequest.Form"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class RequestFormLimitsAttribute : Attribute, IFilterFactory, IOrderedFilter
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="RequestFormLimitsAttribute"/>.
+        /// </summary>
+        public RequestFormLimitsAttribute()
+        {
+            FormOptions = new FormOptions();
+        }
+
+        /// <summary>
+        /// Gets the order value for determining the order of execution of filters. Filters execute in
+        /// ascending numeric value of the <see cref="Order"/> property.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Filters are executed in an ordering determined by an ascending sort of the <see cref="Order"/> property.
+        /// </para>
+        /// <para>
+        /// The default Order for this attribute is 900 because it must run before ValidateAntiForgeryTokenAttribute and
+        /// after any filter which does authentication or login in order to allow them to behave as expected (ie Unauthenticated or Redirect instead of 400).
+        /// </para>
+        /// <para>
+        /// Look at <see cref="IOrderedFilter.Order"/> for more detailed info.
+        /// </para>
+        /// </remarks>
+        public int Order { get; set; } = 900;
```

Comment:
Note that this order number is the same as that of `RequestSizeLimit` filter because both these filters are independent of one another. They just set information on current request's features. Only requirement is for them to run before the antiforgery validation filter (which has an order of 1000)